### ISSUE TITLE
Warning when bluetooth is turned off on OS

### ISSUE THREAD

**Issue by johnnyman727** [sandeepmistry/noble#63]
There should be some kind of warning if there is an attempt to use BLE capabilities when BLE is turned off on a system-wide level.

**Comment by olivierbloch** [comment]
I noticed the following related issue. I have a simple node app that uses noble (on an Intel Edison).
On the noble event 'stateChange', when receiving 'unsupported', I call the system commands to activate BLE ('rfkill unblock bluetooth' and 'hciconfig hci0 up').
I would expect the noble event 'poweredOn' to be triggered but it is not. I have to restart the node app to have noble detect that the BT adapter is now turned on. Shouldn't noble continue monitor the adapter status? Any workaround?

**Comment by sandeepmistry** [comment]
@olivierbloch please open a separate issue to track the behaviour you are seeing on the Intel Edison

**Comment by sandeepmistry** [comment]
Minimal solution for ```noble.startScanning(...)``` had been added in https://github.com/sandeepmistry/noble/commit/5272e058211952a8307587ebda7d74c55d8223d3

If ```callback``` is supplied, it is called with error parameter, otherwise warning is printed to console.

**Comment by ecksun** [comment]
This change causes a race-condition when calling `startScanning` at the beginning of a program:
```
Could not start scanning, state is: unknown (not poweredOn)
cusetup listening at http://0.0.0.0:8080
  noble stateChange poweredOn +0ms
```

Basically, the state is `unknown` when I call startScanning, and slightly after a statechange is emitted (originating from hci-ble I assume) which changes the state.

I was hoping to use this fix to detect the precense of a bluetooth adapter and then noticed that it caused this regression, so I'm not sure if this is the right place to report it or if I should create a new issue for it.

**Comment by sandeepmistry** [comment]
@ecksun this is intended behaviour, you should wait for the adapter state to be determined and ```poweredOn``` before starting scanning.

**Comment by ecksun** [comment]
If thats the case perhaps the documentation should be updated to reflect that, atm the README.md just states to do noble.startScanning() to start scanning.

**Comment by sandeepmistry** [comment]
@ecksun good suggestion, I've incorporated: 7768200

**Comment by ecksun** [comment]
Looks good, but you do realize that this change is a breaking change, so if noble is following semver (I havent figured out if it does or not) then the major version should be increased.
